src/executor/current_thread.rs

```diff
@@ -14,13 +79,25 @@ use std::rc::Rc;
 use std::sync::Arc;
 
 /// Executes tasks on the current thread.
+///
+/// All tasks spawned using this executor will be executed on the current thread
```

Comment:
This is confusing given the inherent methods (which can spawn daemons). Maybe you want to say "when used as an `Executor`?